main.tf

```diff
@@ -182,6 +194,6 @@ resource "aws_cloudwatch_log_subscription_filter" "cloudwatch-to-firehose" {
   destination_arn = aws_kinesis_firehose_delivery_stream.firehose.arn
   filter_pattern  = var.cloudwatch_filter_pattern
   log_group_name  = element(var.cloudwatch_log_group_names, count.index)
-  name            = "firehose-delivery-${element(var.cloudwatch_log_group_names, count.index)}-${random_id.name.hex}"
+  name            = "firehose-delivery-${element(var.cloudwatch_log_group_names, count.index)}-${local.name_unique}"
```

Comment:
You've got me here 😂 `name_prefix` isn't a possible attribute for this resource.